pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1672,6 +1672,20 @@ func (cc *XController) updateQueueJob(oldObj, newObj interface{}) {
 	}
 
 	klog.V(6).Infof("[Informer-updateQJ] '%s/%s' *Delay=%.6f seconds normal enqueue Version=%s Status=%v", newQJ.Namespace, newQJ.Name, time.Now().Sub(newQJ.Status.ControllerFirstTimestamp.Time).Seconds(), newQJ.ResourceVersion, newQJ.Status)
+	for _, cond := range newQJ.Status.Conditions {
+		if cond.Type == arbv1.AppWrapperCondBackoff {
+			//AWs that have backoff conditions have a delay of 10 seconds before getting added to enqueue.
+			//TODO: we could plug an interface here with back-off strategies for different MCAD use cases.
+			time.AfterFunc(time.Duration(cc.serverOption.BackoffTime)*time.Second, func() {
```

Comment:
It'd be useful to have that logic encapsulated into a `enqueueAfter` method, so it mimics the semantic from client-go `DelayingInterface.AddAfter`, and can potentially be reused elsewhere.